resources/providers/awslib/s3/s3.go

```diff
@@ -25,8 +25,18 @@ import (
 )
 
 type BucketDescription struct {
-	Name         string
-	SSEAlgorithm string
+	Name             string
+	SSEAlgorithm     string
+	BucketPolicy     BucketPolicy
+	BucketVersioning BucketVersioning
+}
+
+// TODO: This can be better typed, but this is a complex object. See this library for example: https://github.com/liamg/iamgo/
+type BucketPolicy map[string]any
```

Comment:
Can't we use something similar to [this](https://github.com/primait/nuvola/blob/f373ebd98b16976d6c516fa8fee707fe92b07c48/connector/services/aws/s3/structs.go#L32)?
Type only the relevant fields.

Reply:
IMO it won't be that useful, as you can see in that file most fields are typed as `any`. WDYT?

Reply:
Indeed not very helpful, let's keep it the way it's.